src/Features/Core/Portable/AddImport/AbstractAddImportFeatureService.cs

```diff
@@ -456,10 +441,11 @@ private static HashSet<Project> GetViableUnreferencedProjects(Project project)
             return viableProjects;
         }
 
-        private static void AddRange<TReference>(ArrayBuilder<Reference> allSymbolReferences, ImmutableArray<TReference> proposedReferences, int maxResults)
+        private static void AddRange<TReference>(ConcurrentQueue<Reference> allSymbolReferences, ImmutableArray<TReference> proposedReferences)
```

Comment:
>ConcurrentQueue [](http://example.com/codeflow?start=49&length=15)

Would ConcurrentStack be an option to use here as it has a PushRange method instead of having to Enqueue each item individually?

Reply:
works for me.

Reply:
PushRange takes an array.  Not going to do this :)

Reply:
Works for me. Is there a reason the symbolfinder.Find* methods return immutablearrays?

Reply:
just an easy pattern, and it's nearly always empty.  so not a biggie to address.